cloudferrylib/base/resource.py

```diff
@@ -24,15 +24,16 @@ class Resource(object):
     def __init__(self):
         pass
 
-    def proxy(self, client, cfg):
+    @staticmethod
+    def proxy(client, cfg):
         retry = cfg.migrate.retry
         time_wait = cfg.migrate.time_wait
         return proxy_client.Proxy(client, retry, time_wait)
 
-    def read_info(self, opts=None):
+    def read_info(self, **search_opts):
         pass
 
-    def deploy(self, *args):
+    def deploy(self, info):
```

Comment:
Following implementations will not override this method correctly:

 - cloudferrylib.os.compute.nova_compute.NovaCompute#deploy
 - cloudferrylib.os.object_storage.swift_storage.SwiftStorage#deploy

Let's leave this pylint issue for more global refactoring